### DISCUSSION TITLE
PATCH User email with existing email address

### DISCUSSION THREAD

**Discussion by Smithybrewer** [fastapi-users/fastapi-users#560] [Q&A] [answered]
Just running a few tests and I'm not sure if this warrants creating an issue or whether I'm overlooking something.

If a User updates their email address to a an email address which already exists in the db, rather than getting a "User already exists" response, python throws a UniqueViolationError:

asyncpg.exceptions.UniqueViolationError: duplicate key value violates unique constraint "ix_user_email"
DETAIL:  Key (email)=(user2@example.com) already exists.

I could jump in to the FastAPI-users code and insert a validation in there (essentially copy and paste from the register user flow), but thought I would run it past you first in case it's an error on my behalf or something you might want to be aware of.

Cheers,
Adrian

**Comment by frankie567** [comment, marked as answer]
Indeed, good catch! This is definitely an issue, thanks 👍

**Reply by frankie567** [reply]
Tracked in #561

**Reply by Smithybrewer** [reply]
No worries, glad I could help!
